rabbit-escape-engine/src/rabbitescape/engine/BehaviourTools.java

```diff
@@ -63,6 +63,11 @@ public boolean rabbitIsFalling()
         }
     }
 
+    public boolean rabbitIsClimbing()
+    {
+        return rabbit.state.toString().contains( "CLIMB" );
+    }
+    
```

Comment:
It seems that the tree structure you can get from inheritance seems a natural way to store all those states needed to map to animations. I am interested to see how else you do it.